legal-api/src/legal_api/services/filings/validations/amalgamation_application.py

```diff
@@ -0,0 +1,222 @@
+        if err:
+            msg.extend(err)
+
+    msg.extend(validate_amalgamation_court_order(amalgamation_json, filing_type))
+    msg.extend(validate_amalgamating_businesses(amalgamation_json, filing_type, legal_type, account_id))
+
+    if msg:
+        return Error(HTTPStatus.BAD_REQUEST, msg)
+    return None
+
+
+def validate_amalgamating_businesses(  # pylint: disable=too-many-branches,too-many-statements,too-many-locals
+        amalgamation_json,
+        filing_type,
+        legal_type,
+        account_id) -> list:
+    """Validate amalgamating businesses."""
+    is_staff = jwt.validate_roles([STAFF_ROLE])
+    msg = []
+    amalgamating_businesses_json = amalgamation_json.get('filing', {}) \
+                                                    .get(filing_type, {})\
+                                                    .get('amalgamatingBusinesses', [])
+    amalgamating_businesses_path = f'/filing/{filing_type}/amalgamatingBusinesses'
+    is_any_limited = False
+    is_any_ccc = False
+    is_any_ben = False
+    is_any_ulc = False
+    is_any_expro_a = False
+    amalgamating_businesses = {}
+    for amalgamating_business_json in amalgamating_businesses_json:
+        if identifier := amalgamating_business_json.get('identifier'):
+            business = Business.find_by_identifier(identifier)
+            amalgamating_businesses[identifier] = business
+
+            if business.legal_type == Business.LegalTypes.BCOMP.value:
+                is_any_ben = True
+            elif business.legal_type == Business.LegalTypes.COMP.value:
+                is_any_limited = True
+            elif business.legal_type == Business.LegalTypes.BC_CCC.value:
+                is_any_ccc = True
+            elif business.legal_type == Business.LegalTypes.BC_ULC_COMPANY.value:
+                is_any_ulc = True
+            elif (business and business.legal_type == Business.LegalTypes.EXTRA_PRO_A.value) or \
+                    identifier.startswith('A'):
+                # expro may not be available in lear.checking with identifier prefix as a temporary solution
+                # either enforce affiliation of expro into lear or fetch business from business search
+                is_any_expro_a = True
+    is_any_bc_company = (is_any_ben or is_any_limited or is_any_ccc or is_any_ulc)
+
+    for amalgamating_business_json in amalgamating_businesses_json:
+        identifier = amalgamating_business_json.get('identifier')
+        foreign_legal_name = amalgamating_business_json.get('legalName')
+        is_foreign_business = bool(foreign_legal_name)
+        amalgamating_business = amalgamating_businesses.get(identifier)
+
+        if amalgamating_business and _has_future_effective_filing(amalgamating_business):
+            msg.append({
+                'error': f'{identifier} has a future effective filing',
+                'path': amalgamating_businesses_path
+            })
```

Comment:
so validation fails on any future effective filing on a TING?

Reply:
yes. is in't that how it should be?

Reply:
I haven't looked at that validation requirements document. 

Just wanted to confirm.

Reply:
as per the document it is